avm/res/app/job/tests/e2e/max/main.test.bicep

```diff
@@ -20,6 +20,10 @@ param serviceShort string = 'ajmax'
 @description('Optional. A token to inject into the name of each resource.')
 param namePrefix string = '#_namePrefix_#'
 
+// needed for using listKeys in the secrets, as the storage account is created in the nested deployment and the value needs to exist at the time of deployment
+var storageAccountName = uniqueString('dep-${namePrefix}-menv-${serviceShort}storage')
+var storageAccountId = '/subscriptions/${subscription().subscriptionId}/resourceGroups/${resourceGroupName}/providers/Microsoft.Storage/storageAccounts/${storageAccountName}'
```

Comment:
Can you not use the outputs of the `dependencies.bicep`?

Reply:
Unfortunately that does not work, as the value is needed at "deployment time", which is not after the nested deployment has been executed, but when loading the bicep file.

Reply:
![SmartThinkingGIF](https://github.com/Azure/bicep-registry-modules/assets/5988767/a85e20ea-893e-412c-8785-2985f524e91f)

Reply:
Can't argue with that.

Reply:
Ok, I give up 😄 

@ReneHezser let's go with your solution. But - I'd suggest to change a few small things
1. Let's generate the storage account name only once in the variable & pass it into the `dependencies.bicep`:
   ```bicep
    // needed for using listKeys in the secrets, as the storage account is created in the nested deployment and the value needs to exist at the time of deployment
   var storageAccountName = 'dep${namePrefix}sa${serviceShort}'
   (...)
   module nestedDependencies 'dependencies.bicep' = {
      scope: resourceGroup
      name: '${uniqueString(deployment().name, resourceLocation)}-paramNested'
      params: {
        (...)
       storageAccountName: storageAccountName 
     }
   }
   ```
   Which just requires you to add that input parameter in the `dependencies.bicep` and reference it accordingly. The less places we have to maintain a value, the better. Ideally 1 😉 
2. Let's move the `storageAccountResourceId` variable into the function and shorten it to:
   ` value: listKeys('${resourceGroup.id}/providers/Microsoft.Storage/storageAccounts/${storageAccountName}', '2023-04-01').keys[0].value`
3. Let's cleanup the `dependencies.bicep` outputs and remove the `storageAccountResourceId` as it can unfortunately not be used for what it was intended for

Reply:
Thanks for trying. I know it is not the prettiest solution, but the only working in this case :-(
I updated the code accordingly, except for 2, as I need the storage resource id in two places (and am referring to somebody who said, "keep the places to maintain this to 1".

Reply:
In that case I'd still ague you should shorten it as per `suggestion No. 2` - just without moving it :wink:, sir.

Reply:
Ah, I am sorry. Missed that. I like your suggestion about the shorter id generation.